pkg/ovsdb/database.go

```diff
@@ -209,6 +212,87 @@ func (con *DatabaseEtcd) CreateMonitor(dbName string, handler *Handler, log logr
 	return m
 }
 
+type EventKeyValue struct {
+	Key            string `json:"key"`
+	CreateRevision int64  `json:"create_revision"`
+	ModRevision    int64  `json:"mod_revision"`
+	Version        int64  `json:"version"`
+	Value          string `json:"value"`
+	Lease          int64  `json:"lease"`
+}
+
+func NewEventKeyValue(kv *mvccpb.KeyValue) *EventKeyValue {
+	if kv == nil {
+		return nil
+	}
+	return &EventKeyValue{
+		Key:            string(kv.Key),
+		CreateRevision: kv.CreateRevision,
+		ModRevision:    kv.ModRevision,
+		Version:        kv.Version,
+		Value:          string(kv.Value),
+		Lease:          kv.Lease,
+	}
+}
+
+type Event struct {
+	Type   string         `json:"type"`
+	Kv     *EventKeyValue `json:"kv"`
+	PrevKv *EventKeyValue `json:"prev_kv"`
+}
+
+func NewEvent(ev *clientv3.Event) Event {
+	return Event{
+		Type:   string(ev.Type),
+		Kv:     NewEventKeyValue(ev.Kv),
+		PrevKv: NewEventKeyValue(ev.PrevKv),
+	}
+}
+
+type EventList []Event
+
+func NewEventList(events []*clientv3.Event) EventList {
+	printable := EventList{}
+	for _, ev := range events {
+		if ev != nil {
+			printable = append(printable, NewEvent(ev))
+		}
+	}
+	return printable
+}
+
+func (evList EventList) String() string {
+	b, _ := json.Marshal(evList)
+	return string(b)
+}
+
+type KeyValue struct {
+	Key   common.Key
+	Value map[string]interface{}
+}
+
+func NewKeyValue(etcdKV *mvccpb.KeyValue) (*KeyValue, error) {
+	kv := new(KeyValue)
+
+	/* key */
+	key, err := common.ParseKey(string(etcdKV.Key))
+	if err != nil {
```

Comment:
r u sure if no error that key ptr not nil or better to make sure key != nil before reference it below ?

Reply:
I can add the check, but I doubt that key in an etcd event can be  nil

Reply:
added the checks (just in case)